WordPress/gutenberg
# Enable useSelect autocompletion

## What?

This PR builds `@wordpress/data` types to provide autocompletion support for `useSelect` across the repository.

The two included cosmetic changes in the `block-editor` package are here to showcase how it works:

https://user-images.githubusercontent.com/205419/175293659-4c009b3a-20f0-47e8-b358-0275204d114b.mp4

Note: For now, this only works with the store defined in the same package where `useSelect` is called. With this PR, TypeScript knows how to resolve the type of the `useSelect` hook, but all the cross-package store imports still come across as `any`.

## How

This PR adds a `wordpress__data.d.ts` ambient module that re-exports the types built for the `@wordpress/data` package.

The intention is to keep this change contained to the Gutenberg repository until the types mature more.

Once that happens, the next step would be publishing them by adding `"types": "build-types"` to `package.json` in `@wordpress/data`. It tells TypeScript how to resolve the typing file for `@wordpress/data` ([source](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html))

## Why?

Now that [useSelect is typed in an autocompletion-friendly way](https://github.com/WordPress/gutenberg/pull/41596), it only makes sense to enable that feature across the Gutenberg repository.

## Testing Instructions

Go to `use-available-alignments.js` and confirm the autocompletion works as on the video above.

CC @sarayourfriend @dmsnell @sirreal @jsnajdr @scruffian
